controllers/devbox/internal/controller/devbox_controller.go

```diff
@@ -439,6 +415,255 @@ func (r *DevboxReconciler) getRuntime(ctx context.Context, devbox *devboxv1alpha
 	return runtimecr, nil
 }
 
+func (r *DevboxReconciler) syncNetwork(ctx context.Context, devbox *devboxv1alpha1.Devbox, recLabels map[string]string) error {
+	runtimecr, err := r.getRuntime(ctx, devbox)
+	if err != nil {
+		return err
+	}
+	var servicePorts []corev1.ServicePort
+	for _, port := range runtimecr.Spec.Config.Ports {
+		servicePorts = append(servicePorts, corev1.ServicePort{
+			Name:       port.Name,
+			Port:       port.ContainerPort,
+			TargetPort: intstr.FromInt32(port.ContainerPort),
+			Protocol:   port.Protocol,
+		})
+	}
+	if len(servicePorts) == 0 {
+		servicePorts = []corev1.ServicePort{
+			{
+				Name:       "devbox-ssh-port",
+				Port:       22,
+				TargetPort: intstr.FromInt32(22),
+				Protocol:   corev1.ProtocolTCP,
+			},
+		}
+	}
+	switch devbox.Spec.NetworkSpec.Type {
+	case devboxv1alpha1.NetworkTypeNodePort:
+		return r.syncNodePortNetwork(ctx, devbox, recLabels, servicePorts)
+	case devboxv1alpha1.NetworkTypeWebSocket:
+		return r.syncWebSocketNetwork(ctx, devbox, recLabels, servicePorts)
+	}
+	return nil
+}
+
+func (r *DevboxReconciler) syncWebSocketNetwork(ctx context.Context, devbox *devboxv1alpha1.Devbox, recLabels map[string]string, servicePorts []corev1.ServicePort) error {
+	if err := r.syncPodSvc(ctx, devbox, recLabels, servicePorts); err != nil {
+		return err
+	}
+	if err := r.syncProxyPod(ctx, devbox, recLabels, servicePorts); err != nil {
+		return err
+	}
+	if err := r.syncProxySvc(ctx, devbox, recLabels, servicePorts); err != nil {
+		return err
+	}
+	if err := r.syncProxyIngress(ctx, devbox); err != nil {
+		return err
+	}
+	devbox.Status.Network.Type = devboxv1alpha1.NetworkTypeWebSocket
+	devbox.Status.Network.WebSocket = devbox.Name + "-proxy-ingress"
+	return r.Status().Update(ctx, devbox)
+}
+
+func (r *DevboxReconciler) syncProxyIngress(ctx context.Context, devbox *devboxv1alpha1.Devbox) error {
+	pathType := networkingv1.PathTypePrefix
+	var className = "nginx"
+	ingressPath := []networkingv1.HTTPIngressPath{
+		{
+			Path:     "/",
+			PathType: &pathType,
+			Backend: networkingv1.IngressBackend{
+				Service: &networkingv1.IngressServiceBackend{
+					Name: devbox.Name + "-proxy-svc",
+					Port: networkingv1.ServiceBackendPort{
+						Number: 8080,
+					},
+				},
+			},
+		},
+	}
+
+	ingressSpec := networkingv1.IngressSpec{
+		IngressClassName: &className,
+		Rules: []networkingv1.IngressRule{
+			{
+				Host: devbox.Name + "." + r.HostPath,
```

Comment:
use rand str as domain and set to devbox status. Use status.websocket.proxyDomain if exist.

Reply:
already fix it.